src/movies.js

```diff
@@ -1,22 +1,55 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(movies) {
+  const mappedArray = movies.map(function (singleMovie){
+    return singleMovie.director;
+   })
+   return mappedArray;
+}
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(movies){
+  const movieDirector = movies.filter( (directorName) => {
+    return directorName.director === 'Steven Spielberg';
+  })
+  for (let i=0; movieDirector === 'Steven Spielberg'; i++){
+    const directorDrama = movieDirector.filter( (dramaMovie) => {
+      return dramaMovie.genre === 'Drama';
+    })
+  }
+  return directorDrama;
 
+}
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(movies) {
+  let movieScores = movies.reduce(function (accValue, currentValue){
+    return accValue.score + currentValue.score
+  })
+  return movieScores
 
+}
 // Iteration 4: Drama movies - Get the average of Drama Movies
 function dramaMoviesScore() {}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear() {}
+function orderByYear (movies) {
+  const newArray = movies.map(function (movieYear){
+    return movieYear.year
+   })
+  return newArray.sort();
+}
 
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
-function orderAlphabetically() {}
+function orderAlphabetically(movies) {
+  for (let i=0; i<20;i++){
+    const newArr = movies.map(function(movieTitle){
+     return movieTitle.title
+    })
+    return newArr.sort();
+  }
+}
```

Comment:
Since the task asks you to print first 20values from sorted array, maybe we can first sort and then print? So we actually get the alphabetically first 20 titles ;)

Same as in iteration5, please review how sort method works.